Jellyfin.Server.Implementations/Users/DefaultPasswordResetProvider.cs

```diff
@@ -52,90 +63,96 @@ public class DefaultPasswordResetProvider : IPasswordResetProvider
+                    };
                 }
             }
 
-            if (usersReset.Count < 1)
-            {
-                throw new ResourceNotFoundException($"No Users found with a password reset request matching pin {pin}");
-            }
-
-            return new PinRedeemResult
+            return new CodeRedeemResult
             {
-                Success = true,
-                UsersReset = usersReset.ToArray()
+                Success = false
             };
         }
 
         /// <inheritdoc />
-        public async Task<ForgotPasswordResult> StartForgotPasswordProcess(User user, bool isInNetwork)
+        public async Task<ForgotPasswordResult> StartForgotPasswordProcess(User user)
         {
-            string pin;
+            string code = string.Empty;
             using (var cryptoRandom = RandomNumberGenerator.Create())
             {
-                byte[] bytes = new byte[4];
+                byte[] bytes = new byte[10];
                 cryptoRandom.GetBytes(bytes);
-                pin = BitConverter.ToString(bytes);
+                code = BitConverter.ToString(bytes);
             }
 
-            DateTime expireTime = DateTime.UtcNow.AddMinutes(30);
-            string filePath = _passwordResetFileBase + user.Id + ".json";
-            SerializablePasswordReset spr = new SerializablePasswordReset
+            DateTime expireTime = DateTime.Now.AddMinutes(30);
+            string filePath = _passwordResetFileBase + user.InternalId + ".json";
+
+            SerializablePasswordReset passwordReset = new SerializablePasswordReset
             {
                 ExpirationDate = expireTime,
-                Pin = pin,
-                PinFile = filePath,
-                UserName = user.Username
+                Code = code,
+                File = filePath,
+                Username = user.Username
             };
 
             await using (FileStream fileStream = File.OpenWrite(filePath))
             {
-                _jsonSerializer.SerializeToStream(spr, fileStream);
+                _jsonSerializer.SerializeToStream(passwordReset, fileStream);
                 await fileStream.FlushAsync().ConfigureAwait(false);
             }
 
-            user.EasyPassword = pin;
-            await _userManager.UpdateUserAsync(user).ConfigureAwait(false);
-
-            return new ForgotPasswordResult
+            var response = new ForgotPasswordResult
             {
                 Action = ForgotPasswordAction.PinCode,
-                PinExpirationDate = expireTime,
+                ExpirationDate = expireTime
             };
+
+            _activityManager.Create(new ActivityLog(
+                string.Format(
+                    CultureInfo.InvariantCulture,
+                    _localization.GetLocalizedString("UserPasswordReset"),
+                    user.Username,
+                    code),
+                "UserPolicyUpdated",
+                user.Id));
+
+            return response;
```

Comment:
INFO : Response is not used as a variable. So _activityManager.Create can come before the creation of the response and the response can be directly returned as return new ForgotPasswordResult {...}